src/protocol/http/org/apache/jmeter/protocol/http/curl/BasicCurlParser.java

```diff
@@ -265,134 +232,172 @@ public void setUrl(String url) {
          * @return the headers
          */
         public Map<String, String> getHeaders() {
-            return headers;
+            return Collections.unmodifiableMap(this.headers);
         }
 
+        /**
+         * @return the list of options which are ignored
+         */
         public List<String> getOptionsInProperties() {
-            return optionsInProperties;
+            return Collections.unmodifiableList(this.optionsInProperties);
         }
 
+        /**
+         * @param option the option
+         */
         public void addOptionsInProperties(String option) {
             this.optionsInProperties.add(option);
         }
 
+        /**
+         * @return the maximum transfer rate
+         */
         public int getLimitRate() {
             return limitRate;
         }
 
+        /**
+         * @param limitRate the maximum transfer rate
+         */
         public void setLimitRate(String limitRate) {
             String unit = limitRate.substring(limitRate.length() - 1, limitRate.length()).toLowerCase();
             int value = Integer.parseInt(limitRate.substring(0, limitRate.length() - 1).toLowerCase());
             switch (unit) {
+            case "k":
+                this.limitRate = value * 128;
```

Comment:
Why is the value `128` used for `k`? Before it was `1000` for `m`. My concern was the mixture of base 10 and base 2 for the calculation.
The use of base 2 for the value `8` is OK in any case.

Reply:
The idea of this method is that the user give the bandwith value; and we help them to tranform it to CPS, and return the value in the warning of Http Request, so they can modify it themself in jmeter.properties(374 line) . So this is why the 128 used for 'k'

Reply:
I had two intentions by asking the question. First I wanted to know, why there was a *new* factor on `k`. Second I wanted to point out, that `128` is a *magic constant*. Be explicit on those constants. Use constants with names.
For example that could be written as
```java
private final int ONE_KILOBYTE_IN_CPS = 1024 / 8;
...
this.limitRate = value * ONE_KILOBYTE_IN_CPS;
```
And again we have to decide, whether we want to use the decimal kilo ( `10^3` = `1000`) or the *byte* kilo ( `2^10` = `1024`). As we are emulating curl, we should use their convention.

Reply:
In Curl, the unit of bandwith are  kilobytes(10^3), megabytes(10^6) and gigabytes(10^9) , and in JMeter too. So I think we will use decimal kilo.
[https://blog.e-zest.com/how-to-simulate-network-bandwidth-in-jmeter](https://blog.e-zest.com/how-to-simulate-network-bandwidth-in-jmeter)
[https://unix.stackexchange.com/questions/39218/throttle-the-download-speed-of-wget-or-curl-while-downloading](https://unix.stackexchange.com/questions/39218/throttle-the-download-speed-of-wget-or-curl-while-downloading)